notebooks/sampling/RandomWalk.ipynb

```diff
@@ -0,0 +1,223 @@
+    "| Author Credit |    Date    |  Update      | cuGraph Version |  Test Hardware |\n",
+    "| --------------|------------|--------------|-----------------|----------------|\n",
+    "| Brad Rees     | 04/20/2021 | created      | 0.19            | GV100, CUDA 11.0\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Test Data\n",
+    "We will be using the Zachary Karate club dataset \n",
+    "*W. W. Zachary, An information flow model for conflict and fission in small groups, Journal of\n",
+    "Anthropological Research 33, 452-473 (1977).*\n",
+    "\n",
+    "\n",
+    "![Karate Club](../img/zachary_black_lines.png)\n",
+    "\n",
+    "\n",
+    "Because the test data has vertex IDs starting at 1, the auto-renumber feature of cuGraph (mentioned above) will be used so the starting vertex ID is zero for maximum efficiency. The resulting data will then be auto-unrenumbered, making the entire renumbering process transparent to users."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#  Import the modules\n",
+    "import cugraph\n",
+    "import cudf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read The Data\n",
+    "# Define the path to the test data  \n",
+    "datafile='../data/karate-data.csv'\n",
+    "\n",
+    "gdf = cudf.read_csv(datafile, delimiter='\\t', names=['src', 'dst'], dtype=['int32', 'int32'] )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gdf['wt'] = 1.0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a Graph - using the source (src) and destination (dst) vertex pairs from the Dataframe \n",
+    "G = cugraph.Graph()\n",
+    "G.from_cudf_edgelist(gdf, source='src', destination='dst', edge_attr='wt')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(34, 78)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# some stats on the graph\n",
+    "(G.number_of_nodes(), G.number_of_edges() )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a list with the seeds\n",
+    "seeds = [17,19]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rw, os = cugraph.random_walks(G, seeds, 4)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "the function return two results, let's look at the second one first.  The seed offset, which is a cuDF Series,\n",
```

Comment:
This explanation is awkward.  Perhaps a description of what RandomWalk generates generally would be helpful for some context.  How about:

A random walk generates a path from a seed vertex.  At each step on the random walk (starting from the seed), the random walker picks a random departing edge to traverse.  The random walk will terminate in two situations, when the maximum path length is reached, or when the current vertex on the path has no departing edges to traverse.  The result of a single random walk will be a path of some length less than or equal to the maximum path length.

cugraph.random_walks performs a random walk from each of the specified seeds.  The output will be a path for each of the seeds.  Because the path lengths might be variable length, the return value consists of a pair of outputs.

The first output provides the edges used on the paths.

The second output represents the seed offset, which is a cuDF Series.  The seed offset identifies the offset of the first entry in the first output for a particular seed.

Reply:
used your text.